src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,314 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Component, Directive, Input, Output, EventEmitter,
+    OnDestroy, AfterViewInit, ElementRef, Injectable, Optional} from '@angular/core';
+
+import {Observable} from 'rxjs/Observable';
+import {ScrollDispatcher} from '../core/overlay/scroll/scroll-dispatcher';
+import {Scrollable} from '../core/overlay/scroll/scrollable';
+import {Subject} from 'rxjs/Subject';
+
+
+@Directive({
+    selector: '[cdkStickyRegion]',
+})
+export class StickyParentDirective {
+    constructor(private element: ElementRef) { }
+
+    getElementRef(): ElementRef {
+        return this.element;
+    }
+}
+
+
+@Directive({
+    selector: '[cdkStickyHeader]',
+})
+export class StickyHeaderDirective implements OnDestroy, AfterViewInit {
+
+    /**Set the sticky-header's z-index as 10 in default. Make it as an input
+     * variable to make user be able to customize the zIndex when
+     * the sticky-header's zIndex is not the largest in current page.
+     * Because if the sticky-header's zIndex is not the largest in current page,
+     * it may be sheltered by other element when being sticked.
+     */
+    @Input('sticky-zIndex') zIndex: number = 10;
+    @Input() cdkStickyParentRegion: any;
```

Comment:
' @Input('cdkStickyHeaderParentRegion') parentRegion: HTMLElement '
The parentRegion is a 'HTMLElement'